src/le.py

```diff
@@ -1461,21 +1478,106 @@ def cancel(self):
         if self.timer:
             self.timer.cancel()
 
+class FollowMultilog(object):
+    """
+    The FollowMultilog is responsible for handling those logs that were set-up using the
+    '--multilog' option and that may have a wildcard in the pathname.
+    In which case multiple local (log) files will be followed, but with all the new events
+    from all the files forwarded to the same single log in the logentries infrastructure.
+    """
+
+    def __init__(self, name, entry_filter, entry_formatter, entry_identifier, transport, max_num_followers=MAX_FILES_FOLLOWED ):
+        """ Initializes the FollowMultilog. """
+        self.name = name
+        self.flush = True
+        self.entry_filter = entry_filter
+        self.entry_formatter = entry_formatter
+        self.entry_identifier = entry_identifier
+        self.transport = transport
+
+        self._shutdown = False
+        self._max_num_followers=max_num_followers
+        self._followers = []
+        self._worker = threading.Thread(
+            target=self.start_followers, name=self.name)
+        self._worker.daemon = True
+        self._worker.start()
+
+    def _file_candidates(self):
+        """
+        Return a list of files found for the pathanme passed in
+        need to switch off the globing in each follower!!!!!
+        we put the max here
+        also the dynmaic action will be called from here
+        """
+        ##todo manage dynamic code in this function
+        try:
+            candidates = glob.glob(self.name)
+            if len(candidates) == 0:
+                return None
+            return candidates
+        except os.error:
+            #todo log.debug error
+            return None
+
+    def _file_test(self, candidate):
+        """
+        Only regular files passed
+        """
+        # fail if a symbolic link
+        if os.path.islink(candidate):
+            return False
+        # fail if not a regular file (passes links!)
+        if not os.path.isfile(candidate):
+            return False
+        return True
+
+    def close(self):
+        """
+        Stops all FollowMultilog activity, and then loops through list of existing
+        followers to close each one - then waits for the worker thread to stop.
+        """
+        self._shutdown = True
+        #run through list of followers closing each one
+        for follower in self._followers:
+            follower.close()
+        self._worker.join(1.0)
+
+    def start_followers(self):
+        """
+         Instantiates a Follower object for each file found - all log events from all
+         files are forwarded to the same log in the lE infrastructure
+        """
+        file_count = 0
+        file_candidates = self._file_candidates()
+        # todo log.info self.name
+        for filename in file_candidates:
+            if self._file_test(filename):
+                if file_count < self._max_num_followers:
```

Comment:
I would be more inclined to interpret the max as hard limit. If there is more files I would simply avoid following any. The bottom line here is that the limit is a safety net, nothing else. Now it is prone to race conditions - file listing